### PR TITLE
Doc string enforcer and annotations

### PR DESCRIPTION
**Pull request by pem70**
**What It Does**
Update current doc strings, introduce doc string check, and update annotation for all methods.
[#279] [#309] [#280]

**How to Test**

**Review Checklist**
I certify that I have:
- [x] tested my changes
- [x] added/updated automated tests
- [x] updated the changelog
- [x] followed the [contribution guidelines](https://github.com/zowe/zowe-cli/blob/master/CONTRIBUTING.md)

**Additional Comments**

### CODE REVIEW

File: src/zos_files/zowe/zos_files_for_zowe_sdk/uss.py [hunk 1]
```diff
@@ -145,15 +173,42 @@ def get_content_streamed(self, file_path, binary=False):
...
         response = self.get_content_streamed(file_path, binary)
         with open(output_file, "wb" if binary else "w", encoding="utf-8") as f:
             for chunk in response.iter_content(chunk_size=4096, decode_unicode=not binary):
                 f.write(chunk)
 
-    def upload(self, input_file, filepath_name, encoding=_ZOWE_FILES_DEFAULT_ENCODING):
-        """Upload contents of a given file and uploads it to UNIX file"""
+    def upload(self, input_file: str, filepath_name: str, encoding: str = _ZOWE_FILES_DEFAULT_ENCODING):
+        """
+        Upload contents of a given file and uploads it to UNIX file.
```
Comment (traeok) [inline src/zos_files/zowe/zos_files_for_zowe_sdk/uss.py:196]: Thoughts on updating this description for clarity?

```suggestion
        Uploads contents of a given file and saves it to a file at the given USS path.
```

File: src/zos_files/zowe/zos_files_for_zowe_sdk/uss.py [hunk 2]
```diff
@@ -117,26 +131,40 @@ def write(self, filepath_name, data, encoding=_ZOWE_FILES_DEFAULT_ENCODING):
...
+            Path of the file
+        binary: bool
+            Specifies whether the contents are binary
 
         Returns
         -------
-        response
-            A response object from the requests library
+        dict
+            A JSON response of results of the operaiotn
```
Comment (traeok) [inline src/zos_files/zowe/zos_files_for_zowe_sdk/uss.py:167]: Line 167 - "operaiotn" -> "operation"

```suggestion
            A JSON response with results of the operation
```

File: src/core/zowe/core_for_zowe_sdk/session.py [hunk 3]
```diff
@@ -85,9 +92,25 @@ def __init__(self, props: dict) -> None:
...
     @property
     def host_url(self) -> str:
+        """
+        Return the fommatted host url.
+
+        Returns
+        -------
+        str
+            the fommatted host url
+        """
```
Comment (traeok) [inline src/core/zowe/core_for_zowe_sdk/session.py:114]:
```suggestion
        """
        Return the formatted host URL.

        Returns
        -------
        str
            the formatted host URL
        """
```

File: src/core/zowe/core_for_zowe_sdk/profile_manager.py [hunk 4]
```diff
@@ -253,8 +333,34 @@ def load(
...
         If `profile_type` is not base, then we will load properties from both
         `profile_type` and base profiles and merge them together.
-        """
 
+        Parameters
+        ----------
+        profile_name : Optional[str]
+            The name of the profile to load. If None, profiles are loaded based only on profile type.
+        profile_type : Optional[str]
+            The type of the profile to load, e.g., 'user', 'admin'. If None, profiles are loaded based only on name.
```
Comment (traeok) [inline src/core/zowe/core_for_zowe_sdk/profile_manager.py:342]: Do we want to provide valid examples for the profile types to load?

```suggestion
            The type of the profile to load, e.g., 'zosmf', 'zftp'. If None, profiles are loaded based only on name.
```

File: requirements.txt [hunk 5]
```diff
@@ -29,11 +30,12 @@ typing-extensions==4.0.1
 Unidecode==1.2.0
 urllib3==1.26.19
 wheel
-zipp==3.19.1
+zipp==3.4.0
```
Comment (traeok) [inline requirements.txt:33]: Was this change required to add `pydoclint`?

**Review by traeok — COMMENTED** [review]
Looks good so far, thanks @pem70! Left a few comments, mostly regarding the changes in `requirements.txt` and some of the doc strings.

File: src/zos_files/zowe/zos_files_for_zowe_sdk/uss.py (hunk 1 above)
Comment (pem70) [reply src/zos_files/zowe/zos_files_for_zowe_sdk/uss.py]: I agree with you. The revision is employed manually since numpy style in Pydoclint requires the first sentence to be imperative mood (No following 's' for verbs).

File: requirements.txt (hunk 5 above)
Comment (pem70) [reply requirements.txt]: It is not required and has been revised back to zipp==3.19.1. I wonder how this happened.

**Review by traeok — APPROVED** [review]
LGTM, thanks @pem70

File: CHANGELOG.md [hunk 6]
```diff
@@ -8,6 +8,14 @@ All notable changes to the Zowe Client Python SDK will be documented in this fil
 
 - Included support for `AUTH_TYPE_CERT_PEM` and `AUTH_TYPE_NONE` in `session` [#291] (https://github.com/zowe/zowe-client-python-sdk/issues/291) and [#296] (https://github.com/zowe/zowe-client-python-sdk/issues/296)
 
+- Updated all functions descriptions to be consitent [#279] (https://github.com/zowe/zowe-client-python-sdk/issues/279)
```
Comment (t1m0thyj) [inline CHANGELOG.md:11]:
```suggestion
- Updated doc strings for all functions to be consistent [#279] (https://github.com/zowe/zowe-client-python-sdk/issues/279)
```

File: src/zos_files/zowe/zos_files_for_zowe_sdk/datasets.py [hunk 7]
```diff
@@ -511,12 +630,22 @@ def get_binary_content(self, dataset_name, stream=False, with_prefixes=False):
...
+        Write content to an existing dataset.
+
+        Parameters
+        ----------
+        dataset_name: str
+            Name of the dataset to retrieve
+        data: str
+            Content to be written
+        encoding: str
+            Specifies encoding schema
```
Comment (t1m0thyj) [inline src/zos_files/zowe/zos_files_for_zowe_sdk/datasets.py:644]:
```suggestion
            Specifies encoding name (e.g. IBM-1047)
```

File: src/zos_files/zowe/zos_files_for_zowe_sdk/datasets.py [hunk 8]
```diff
@@ -526,30 +655,62 @@ def write(self, dataset_name, data, encoding=_ZOWE_FILES_DEFAULT_ENCODING):
...
-        dataset_name:str - Name of the dataset to download
-        output_file:str - Name of the local file to create
-        with_prefixes:boolean - If true, include a four big endian bytes record length prefix.
-                                The default is False
+        dataset_name : str
+            Name of the dataset to download
+        output_file : str
+            Name of the local file to create
+        with_prefixes : bool
+            If true, include a four big endian bytes record length prefix. The default is False
```
Comment (t1m0thyj) [inline src/zos_files/zowe/zos_files_for_zowe_sdk/datasets.py:685]: Should we use formatting consistent with `get_binary_content`?
```suggestion
            If true, include a four big endian bytes record length prefix. Default: False
```

File: src/zos_files/zowe/zos_files_for_zowe_sdk/datasets.py [hunk 9]
```diff
@@ -526,30 +655,62 @@ def write(self, dataset_name, data, encoding=_ZOWE_FILES_DEFAULT_ENCODING):
...
+        Upload contents of a given file and uploads it to a dataset.
+
+        Parameters
+        ----------
+        input_file: str
+            Name of the file to be uploaded
+        dataset_name: str
+            Name of the dataset to be created
+        encoding: str
+            Specifies the encoding schema
```
Comment (t1m0thyj) [inline src/zos_files/zowe/zos_files_for_zowe_sdk/datasets.py:705]:
```suggestion
            Specifies the encoding name (e.g. IBM-1047)
```

File: src/zos_files/zowe/zos_files_for_zowe_sdk/file_system.py [hunk 10]
```diff
@@ -18,42 +20,42 @@
...
 
-        Parameter
-        ---------
-        file_system_name: str - the name for the file system
+        Parameters
+        ----------
+        file_system_name: str
+            the name for the file system
+        options: dict
+            specifies file system attributes
```
Comment (t1m0thyj) [inline src/zos_files/zowe/zos_files_for_zowe_sdk/file_system.py:46]: I think we should use sentence case to make the descriptions consistent
```suggestion
        file_system_name: str
            Name of the file system
        options: dict
            Specifies file system attributes
```

File: src/zos_files/zowe/zos_files_for_zowe_sdk/file_system.py [hunk 11]
```diff
@@ -72,26 +74,46 @@ def create(self, file_system_name, options={}):
...
+        Parameters
+        ----------
+        file_system_name: str
+            Name for the file system
+        mount_point: str
+            Mount point to be used for mounting the UNIX file system
+        options: dict
+            A JSON of request body options
+        encoding: str
+            Specifies optional encoding schema
```
Comment (t1m0thyj) [inline src/zos_files/zowe/zos_files_for_zowe_sdk/file_system.py:111]:
```suggestion
            Specifies optional encoding name (e.g. IBM-1047)
```

File: src/zos_files/zowe/zos_files_for_zowe_sdk/file_system.py [hunk 12]
```diff
@@ -102,17 +124,23 @@ def mount(self, file_system_name, mount_point, options={}, encoding=_ZOWE_FILES_
...
-        file_system_name: str - the name for the file system
-        options: dict - A JSON of request body options
+        Parameters
+        ----------
+        file_system_name: str
+            Name for the file system
+        options: dict
+            A JSON of request body options
+        encoding: str
+            Specifies optional encoding schema
```
Comment (t1m0thyj) [inline src/zos_files/zowe/zos_files_for_zowe_sdk/file_system.py:138]:
```suggestion
            Specifies optional encoding name (e.g. IBM-1047)
```

File: src/zos_files/zowe/zos_files_for_zowe_sdk/file_system.py [hunk 13]
```diff
@@ -122,19 +150,24 @@ def unmount(self, file_system_name, options={}, encoding=_ZOWE_FILES_DEFAULT_ENC
...
-        Parameter
-        ---------
-        file_path: str - the UNIX directory that contains the files and directories to be listed.
-        file_system_name: str - the name for the file system to be listed
+        Parameters
+        ----------
+        file_path_name: Optional[str]
+            the UNIX directory that contains the files and directories to be listed.
+        file_system_name: Optional[str]
+            the name for the file system to be listed
```
Comment (t1m0thyj) [inline src/zos_files/zowe/zos_files_for_zowe_sdk/file_system.py:165]: Same comment as above about sentence case for consistency
```suggestion
        file_path_name: Optional[str]
            USS directory that contains the files and directories to be listed
        file_system_name: Optional[str]
            Name of the file system to be listed
```

File: src/zos_files/zowe/zos_files_for_zowe_sdk/uss.py [hunk 14]
```diff
@@ -60,21 +56,21 @@ def list(self, path):
...
-        Delete a file or directory
+        Delete a file or directory.
 
         Parameters
         ----------
-        filepath of the file to be deleted
-
-        recursive
+        filepath_name: str
+            filepath of the file to be deleted
```
Comment (t1m0thyj) [inline src/zos_files/zowe/zos_files_for_zowe_sdk/uss.py:66]:
```suggestion
            Path of the file to be deleted
```

File: src/zos_files/zowe/zos_files_for_zowe_sdk/uss.py [hunk 15]
```diff
@@ -84,17 +80,24 @@ def delete(self, filepath_name, recursive=False):
...
         ----------
-        file_path of the file to add
-        type = "file" or "dir"
-        mode Ex:- rwxr-xr-x
+        file_path: str
+            file_path of the file to add
+        type: str
+            "file" or "dir"
+        mode: Optional[str]
+            Ex:- rwxr-xr-x
```
Comment (t1m0thyj) [inline src/zos_files/zowe/zos_files_for_zowe_sdk/uss.py:94]:
```suggestion
        file_path: str
            Path of the file to add
        type: str
            Specify either "file" or "dir"
        mode: Optional[str]
            Unix permission string (e.g. `rwxr-xr-x`)
```

File: src/zos_files/zowe/zos_files_for_zowe_sdk/uss.py [hunk 16]
```diff
@@ -103,11 +106,22 @@ def create(self, file_path, type, mode=None):
...
+        Write content to an existing UNIX file.
+
+        Parameters
+        ----------
+        filepath_name: str
+            Path of the file
+        data: str
+            Contents to be written
+        encoding: str
+            Specifies the encoding schema
```
Comment (t1m0thyj) [inline src/zos_files/zowe/zos_files_for_zowe_sdk/uss.py:120]:
```suggestion
            Specifies the encoding name (e.g. IBM-1047)
```

File: src/zos_files/zowe/zos_files_for_zowe_sdk/datasets.py [hunk 17]
```diff
@@ -273,12 +338,33 @@ def list(self, name_pattern, return_attributes=False):
...
+        dataset_name: str
+            The name of the dataset
+        member_pattern: Optional[str]
+            Filters members by name pattern.
+        member_start: Optional[str]
+            The starting point for listing members.
+        limit: int
+            The maximum number of members returned.
+        attributes: str
+            The member attributes to retrieve.
```
Comment (t1m0thyj) [inline src/zos_files/zowe/zos_files_for_zowe_sdk/datasets.py:363]: Trailing periods should be removed for consistency

File: src/zos_files/zowe/zos_files_for_zowe_sdk/datasets.py [hunk 18]
```diff
@@ -296,41 +382,47 @@ def list_members(self, dataset_name, member_pattern=None, member_start=None, lim
...
+        alias: Optional[bool]
             Alias of the dataset to copy from
-        to_member_name: str
+        to_member_name: Optional[str]
             Name of the member to copy to
-        enq: str
+        enq: Optional[str]
             Enqueue type for the dataset to copy from
         replace: bool
             If true, members in the target data set are replaced.
```
Comment (t1m0thyj) [inline src/zos_files/zowe/zos_files_for_zowe_sdk/datasets.py:414]: Trailing periods should be removed for consistency

File: src/zos_files/zowe/zos_files_for_zowe_sdk/datasets.py [hunk 19]
```diff
@@ -474,32 +581,44 @@ def create_default(self, dataset_name: str, default_type: str):
...
+    def get_content(self, dataset_name: str, stream: bool = False) -> dict:
+        """
+        Retrieve the contents of a given dataset.
+
+        Parameters
+        ----------
+        dataset_name: str
+            The name of the dataset
+        stream: bool
+            Specifies whether the request is streaming
```
Comment (t1m0thyj) [inline src/zos_files/zowe/zos_files_for_zowe_sdk/datasets.py:593]: Should we include the default value?
```suggestion
            Specifies whether the response is streamed. Default: False
```

**Review by t1m0thyj — CHANGES_REQUESTED** [review]
Looks pretty good, thanks @pem70!

Left minor comments that are unresolved from #307

**Comment by zFernand0** [comment]
When resolving these merge conflicts, remember to remove the `basicConfig` setup (removed in #313)

**Review by zFernand0 — APPROVED** [review]

File: src/zos_files/zowe/zos_files_for_zowe_sdk/datasets.py [hunk 20]
```diff
@@ -213,96 +233,99 @@ def blksize(self, blksize: Optional[int]):
...
     def like(self) -> Optional[str]:
+        """Get the dataset name to copy attributes from."""
         return self.__like
 
-    def to_dict(self) -> dict:
-        """Return the DatasetOption as a dict
+    @like.setter
+    def like(self, like: Optional[str]):
+        """Set the dataset name to copy attributes from."""
+        self.__like = like
```
Comment (t1m0thyj) [inline src/zos_files/zowe/zos_files_for_zowe_sdk/datasets.py:292]: Curious why we're adding a setter for `like` in this PR? I believe that beforehand we decided against having a setter for it, because it would require reinitializing the class to update default values of other properties.

**Review by t1m0thyj — APPROVED** [review]
LGTM, thanks @pem70!
